tutoraspects/templates/aspects/apps/superset/pythonpath/superset_config.py

```diff
@@ -151,6 +151,19 @@ class CeleryConfig(object):
 BABEL_DEFAULT_LOCALE = "{{ SUPERSET_DEFAULT_LOCALE }}"
 LANGUAGES = {{ SUPERSET_SUPPORTED_LANGUAGES }}
 
+def DB_CONNECTION_MUTATOR(uri, params, username, security_manager, source):
```

Comment:
This is where the actual magic happens. The ClickHouse username / password is replaced dynamically with the one we create for them at query time.